### PR TITLE
[Enhancement] Enables SR cluster can support IP and FQDN conversion

### LINKED ISSUE #10995: Enables SR cluster can rollback from using FQDN to IP

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #10995 

## Problem Summary(Required) ：

Previously SR cluster can‘t change FQDN to IP, this PR Enables SR cluster can support IP and FQDN conversion
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java|7|7|100.00%|[]|

File: be/src/agent/heartbeat_server.cpp [hunk 1]
```diff
@@ -147,13 +147,18 @@ StatusOr<HeartbeatServer::CmpResult> HeartbeatServer::compare_master_info(const
...
 
-            std::string ip = hostname_to_ip(master_info.backend_ip);
-            if (ip.empty()) {
-                return Status::InternalError("can not get ip from fqdn");
+            std::string ip;
+            if (is_valid_ip(master_info.backend_ip)) {
+                ip = master_info.backend_ip;
+            } else {
+                ip = hostname_to_ip(master_info.backend_ip);
+                if (ip.empty()) {
```
Comment (chaoyli) [inline be/src/agent/heartbeat_server.cpp:159]: please print the IP or FQDN in the status

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java|7|7|100.00%|[]|

File: be/src/agent/heartbeat_server.cpp (hunk 1 above)
Comment (waittttting) [reply be/src/agent/heartbeat_server.cpp:160]: done

**Comment by waittttting** [comment]
run starrocks_clang-tidy

**Review by chaoyli — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
backport branch-2.4

**Comment by waittttting** [comment]
https://github.com/Mergifyio backport branch-2.4

**Pull request merged**